### PR TITLE
Settings api

### PR DESCRIPTION
**Pull request by paytonshaltis**
This merge is a joint effort between front and backend code:

- The settings page displays the current user settings.
- The settings page allows these settings to be changed.
- The backend API receives the request and updates the settings in the database.
- The user's session and other frontend pages are updated with new values.

This PR also closes multiple smaller issues that I will link. Feel free to try modifying settings for yourself. There are a few minor changes that I will make a separate issue about, but for the most part this page is functional and does what it needs to.

### CODE REVIEW

File: views/account/settings/index.ejs [hunk 1]
```diff
@@ -9,41 +9,144 @@
...
+
+      <div class="form-group row">
+        <label for="inputFontSize" class="col-sm-2 col-form-label"
+          >Font Size:</label
+        >
+        <div class="col-sm-2">
+          <div class="form-group">
+            <select name="font_size" class="form-control" id="inputFontSize">
+              <!-- Adds the options and selects the user's current setting. -->
+              <% options.font_size.forEach( (option) => { if (settings.font_size
```
Comment (paytonshaltis) [inline views/account/settings/index.ejs:69]: I know that this way is a bit tedious, but EJS allows us to use values directly without having to make a separate request in the main.js file. I'm creating an issue that looks into how complex making the change would be.

File: views/account/settings/index.ejs [hunk 2]
```diff
@@ -9,41 +9,144 @@
...
-            aria-disabled="true"
-            >Generate</a
-          >
-        </main>
-      </div>
+    <!-- Parse the user's settings as JSON before using. -->
+    <% var settings = JSON.parse(user.settings); %>
+
+    <!-- Create the options. -->
+    <%var options = { font_size: ["12pt", "14pt", "16pt", "18pt"], font_family:
```
Comment (paytonshaltis) [inline views/account/settings/index.ejs:16]: These options are also not set in stone... if anyone has ideas feel free to share them with the group.

Comment (sderacy) [reply views/account/settings/index.ejs:16]: The font size options should be chosen with consideration for how they look on the webpage. We should make sure the largest, 18pt font, wouldn't alter the page in any serious way

Comment (paytonshaltis) [reply views/account/settings/index.ejs:16]: Agreed. I also think more fonts may emerge once we start user testing, but the code is currently flexible enough to handle changing these.

**Review by sderacy — APPROVED** [review]
LGTM

**Pull request merged**